crypto/pem/pem_lib.c

```diff
@@ -416,6 +416,106 @@ int PEM_ASN1_write_bio(i2d_of_void *i2d, const char *name, BIO *bp,
     return ret;
 }
 
+int PEM_ASN1_write_bio_ctx(OSSL_i2d_of_void_ctx *i2d, void *vctx,
+                           const char *name, BIO *bp, const void *x,
+                           const EVP_CIPHER *enc, const unsigned char *kstr,
+                           int klen, pem_password_cb *callback, void *u)
+{
```

Comment:
Maybe there is a nicer way to refactor this to reuse code segments between the 2 variants of the function. They seem to only differ in the i2d calls.
e.g. doing conditional calls in the _ctx variant and making PEM_ASN1_write_bio() just a stub that calls the new method with NULL passed as the ctx?

Reply:
The NULL could be a valid ctx, required by some of the `i2d` methods. There'd need to a separate `i2d_with_ctx` argument for a common static function, with each of the two passing either a non-NULL traditional `i2d` or a non-NULL three-argument `i2d` pointer. Then they can share code.